identity/scim.go

```diff
@@ -45,6 +46,13 @@ type UserPatchOperations struct {
 	Value *GroupsValue `json:"value,omitempty"`
 }
 
+// ServicePrincipalPatchOperations is a list of path operations for add or removing service principal attributes
+type ServicePrincipalPatchOperations struct {
```

Comment:
there's no need to create a dedicated `*PatchOperations` struct, you can reuse the user one.

Reply:
sounds good, will change it, but should we rename the struct of UserPatchOperation to make it more abstract?

Reply:
ad0a9e4065935cd6cda6064ce5ad45ca445f9bac

Reply:
@tcz001 yes, it can be just PathOperations